azkaban-common/src/main/java/azkaban/jobExecutor/AbstractProcessJob.java

```diff
@@ -81,6 +84,31 @@ protected void resolveProps() {
     jobProps = PropsUtils.resolveProps(jobProps);
   }
 
+  /**
+   * prints the current Job props to the Job log.
+   */
+  protected void logJobProperties() {
+    if (this.jobProps != null &&
+        this.jobProps.getBoolean(JOB_DUMP_PROPERTIES_IN_LOG, false)){
+      try {
+        Map<String,String> flattenedProps = this.jobProps.getFlattened();
```

Comment:
yep, it is a great idea to pull the keySet here as the implamentation does go recursively and gets all the keys from the hierarchy chain, however the concern for this implementation comes from the code to pull the value from the key, it seems like the props object only exposes  the function get(object) to fetch the value from the key, which furtunately does have the logic to go all the way up the chain when finding the value, however as the way it implements it need to go thru the chain each time to find a value for the key, it will be expensive if a value from the top parent is pulled for many times with this implementation. I agree the keySet route does have the benefit to avoid touching the props class, however I feel it might be nicer to provide this extra method, which caches the flatted key value pairs,  to fulfill the functionality demand of the feature and avoid potential performance draw backs.  Please let me know your idea. (btw, seems like jobProps.getkeySet returns only a HashSet, so it seems like we need to explictly initialize a TreeSet from the return of the getKeySet() call rather than assigning directly, let me know if I got it wrong :)  )